lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -395,6 +396,13 @@ bool TypeLoweringVisitor::lowerProducer(
     ArrayAttr loweredAttrs =
         filterAnnotations(context, oldAnno, srcType, field);
     auto newOp = clone(field, loweredName, loweredAttrs);
+    // Carry over the inner_sym name, if present.
+    if (innerSym) {
+      if (!lowered.empty())
+        op->emitError("replication due to type lowering renders @")
```

Comment:
I think the one case we will care about is when a path (which has a pointer to this symbol) resolves to an individual element of the expanded thing, in which case it's not ambiguous, but it does require looking through the users.
This doesn't need to happen now, but I think this will come up soon.

Reply:
Yeah good point. This breaking upon `LowerTypes` seems awfully brittle in general, and fleshing out some proper semantics on what should happen with symbols here would be great!